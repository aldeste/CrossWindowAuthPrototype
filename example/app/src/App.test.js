import React from "react";
import App from "./App";
import renderer from "react-test-renderer";

jest.mock("./Document/Title", () => ({ children }) => (
  <title>{children}</title>
));
// Remove all class names from tags as they're auto generated,
// will cause tests to fail
jest.mock("./Tags/Button", () => ({ className, children, ...props }) => (
  <button {...props}>{children}</button>
));
jest.mock("./Tags/Text", () => ({ className, children, ...props }) => (
  <p {...props}>{children}</p>
));
jest.mock("./Tags/Title", () => ({ className, children, ...props }) => (
  <h1 {...props}>{children}</h1>
));
jest.mock("./Tags/Form", () => ({ className, children, ...props }) => (
  <form {...props}>{children}</form>
));
jest.mock("./Tags/Label", () => ({ className, children, ...props }) => (
  <label {...props}>{children}</label>
));
jest.mock("./Tags/TextInput", () => ({ className, children, ...props }) => (
  <input {...props}>{children}</input>
));
jest.mock("./Tags/View", () => ({ className, children, ...props }) => (
  <div {...props}>{children}</div>
));

it("renders without crashing", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Updates satate to reflect login", () => {
  const component = renderer.create(<App />);
  component.getInstance().signIn("Star Wars")({ username: "Yoda" });
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
