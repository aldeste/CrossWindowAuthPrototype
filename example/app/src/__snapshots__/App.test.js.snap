// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application loads asynchronously loads in the welcome screen asynchronously 1`] = `
<div>
  <title>
    AuthJazz
  </title>
  <p>
    {
      "StarWars": {
        "name": "Yoda"
      }
    }
  </p>
  <div>
    <h1>
      Connection testing field
    </h1>
    <button
      alternative={true}
      onClick={[Function]}
      outlined={true}
    >
      Click to ping
    </button>
  </div>
  <div>
    <h1>
      Star Wars Characters login
    </h1>
    <p>
      Please log in
    </p>
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="StarWarsCharactersusername"
          required={false}
        >
          Username
        </label>
        <input
          id="StarWarsCharactersusername"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          value=""
        />
      </div>
      <div>
        <label
          htmlFor="StarWarsCharacterspassword"
          required={false}
        >
          Password
        </label>
        <input
          id="StarWarsCharacterspassword"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <button
        disabled={true}
      >
        Form fields not valid
      </button>
    </form>
  </div>
  <p>
    What follows is an iframe, the colors are different to illustrate that we're accessing a different website
  </p>
  <div
    src="http://localhost:4050"
    title="application_display"
  />
</div>
`;

exports[`Application loads asynchronously starts of empty 1`] = `
<div>
  <title>
    AuthJazz
  </title>
  <p>
    What follows is an iframe, the colors are different to illustrate that we're accessing a different website
  </p>
  <div
    src="http://localhost:4050"
    title="application_display"
  />
</div>
`;

exports[`Application start file Logs in users by token 1`] = `
<div>
  <title>
    AuthJazz
  </title>
  <div>
    <h1>
      Connection testing field
    </h1>
    <button
      alternative={true}
      onClick={[Function]}
      outlined={true}
    >
      Click to ping
    </button>
  </div>
  <div>
    <h1>
      Star Wars login
    </h1>
    <p>
      Please log in
    </p>
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="StarWarsusername"
          required={false}
        >
          Username
        </label>
        <input
          id="StarWarsusername"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          value=""
        />
      </div>
      <div>
        <label
          htmlFor="StarWarspassword"
          required={false}
        >
          Password
        </label>
        <input
          id="StarWarspassword"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <button
        disabled={true}
      >
        Form fields not valid
      </button>
    </form>
  </div>
  <div>
    <h1>
      Star Wars Characters login
    </h1>
    <p>
      Please log in
    </p>
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="StarWarsCharactersusername"
          required={false}
        >
          Username
        </label>
        <input
          id="StarWarsCharactersusername"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          value=""
        />
      </div>
      <div>
        <label
          htmlFor="StarWarsCharacterspassword"
          required={false}
        >
          Password
        </label>
        <input
          id="StarWarsCharacterspassword"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <button
        disabled={true}
      >
        Form fields not valid
      </button>
    </form>
  </div>
  <p>
    What follows is an iframe, the colors are different to illustrate that we're accessing a different website
  </p>
  <div
    src="http://localhost:4050"
    title="application_display"
  />
</div>
`;

exports[`Application start file Removes the welcome screen on logout 1`] = `
<div>
  <title>
    AuthJazz
  </title>
  <div>
    <h1>
      Connection testing field
    </h1>
    <button
      alternative={true}
      onClick={[Function]}
      outlined={true}
    >
      Click to ping
    </button>
  </div>
  <div>
    <h1>
      Star Wars login
    </h1>
    <p>
      Please log in
    </p>
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="StarWarsusername"
          required={false}
        >
          Username
        </label>
        <input
          id="StarWarsusername"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          value=""
        />
      </div>
      <div>
        <label
          htmlFor="StarWarspassword"
          required={false}
        >
          Password
        </label>
        <input
          id="StarWarspassword"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <button
        disabled={true}
      >
        Form fields not valid
      </button>
    </form>
  </div>
  <div>
    <h1>
      Star Wars Characters login
    </h1>
    <p>
      Please log in
    </p>
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="StarWarsCharactersusername"
          required={false}
        >
          Username
        </label>
        <input
          id="StarWarsCharactersusername"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          value=""
        />
      </div>
      <div>
        <label
          htmlFor="StarWarsCharacterspassword"
          required={false}
        >
          Password
        </label>
        <input
          id="StarWarsCharacterspassword"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <button
        disabled={true}
      >
        Form fields not valid
      </button>
    </form>
  </div>
  <p>
    What follows is an iframe, the colors are different to illustrate that we're accessing a different website
  </p>
  <div
    src="http://localhost:4050"
    title="application_display"
  />
</div>
`;

exports[`Application start file Renders the welcome screen on login 1`] = `
<div>
  <title>
    AuthJazz
  </title>
  <p>
    {
      "StarWars": {
        "name": "Yoda"
      }
    }
  </p>
  <div>
    <h1>
      Connection testing field
    </h1>
    <button
      alternative={true}
      onClick={[Function]}
      outlined={true}
    >
      Click to ping
    </button>
  </div>
  <div>
    <h1>
      Star Wars user logged in!
    </h1>
    <p>
      You're logged in as 
      Yoda
    </p>
    <button
      alternative={true}
      onClick={[Function]}
    >
      Log out
    </button>
  </div>
  <div>
    <h1>
      Star Wars Characters login
    </h1>
    <p>
      Please log in
    </p>
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="StarWarsCharactersusername"
          required={false}
        >
          Username
        </label>
        <input
          id="StarWarsCharactersusername"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          value=""
        />
      </div>
      <div>
        <label
          htmlFor="StarWarsCharacterspassword"
          required={false}
        >
          Password
        </label>
        <input
          id="StarWarsCharacterspassword"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <button
        disabled={true}
      >
        Form fields not valid
      </button>
    </form>
  </div>
  <p>
    What follows is an iframe, the colors are different to illustrate that we're accessing a different website
  </p>
  <div
    src="http://localhost:4050"
    title="application_display"
  />
</div>
`;

exports[`Application start file renders without crashing 1`] = `
<div>
  <title>
    AuthJazz
  </title>
  <div>
    <h1>
      Connection testing field
    </h1>
    <button
      alternative={true}
      onClick={[Function]}
      outlined={true}
    >
      Click to ping
    </button>
  </div>
  <div>
    <h1>
      Star Wars login
    </h1>
    <p>
      Please log in
    </p>
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="StarWarsusername"
          required={false}
        >
          Username
        </label>
        <input
          id="StarWarsusername"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          value=""
        />
      </div>
      <div>
        <label
          htmlFor="StarWarspassword"
          required={false}
        >
          Password
        </label>
        <input
          id="StarWarspassword"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <button
        disabled={true}
      >
        Form fields not valid
      </button>
    </form>
  </div>
  <div>
    <h1>
      Star Wars Characters login
    </h1>
    <p>
      Please log in
    </p>
    <form
      onSubmit={[Function]}
    >
      <div>
        <label
          htmlFor="StarWarsCharactersusername"
          required={false}
        >
          Username
        </label>
        <input
          id="StarWarsCharactersusername"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          value=""
        />
      </div>
      <div>
        <label
          htmlFor="StarWarsCharacterspassword"
          required={false}
        >
          Password
        </label>
        <input
          id="StarWarsCharacterspassword"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <button
        disabled={true}
      >
        Form fields not valid
      </button>
    </form>
  </div>
  <p>
    What follows is an iframe, the colors are different to illustrate that we're accessing a different website
  </p>
  <div
    src="http://localhost:4050"
    title="application_display"
  />
</div>
`;
